### DISCUSSION TITLE
Planning for v0.16 Release

### DISCUSSION THREAD

**Discussion by mikeingold** [mikeingold/LineIntegrals.jl#142] [General]
### User-Facing Changes
- Added a `diff_method` keyword argument to `integral` and implemented `DifferentiationMethod` types for specification: `FiniteDifference` approximations and `AutoEnzyme` for Enzyme.jl AD use.
- Changed `jacobian` and `differential` signature to `(geometry, ts[, ::DifferentiationMethod])`
- Deprecated usage of `GaussKronrod` rules for surfaces.
- Adds support for integrating `Tetrahedron` like any other volume.
- Generalizes integrand functions to support any `f::Any` with a method defined for `f(::Point)`.
- Performance Improvements
  - Achieved an 80x improvement when integrating over `BezierCurve`.
  - Achieved an up-to-4x improvement when integrating using `HAdaptiveCubature`.
- improved documentation system.
  - Updated documentation to explain usage of differential forms and why specialization methods exist.
  - Use `@autodocs` to automatically add all docstrings to API page.

### Internal Changes
- Implemented a `_ParametricGeometry <: Meshes.Geometry` for defining geometries with custom parametric functions, and used it to implement specialization methods for
  - `BezierCurve`
  - `Line`
  - `Plane`
  - `Ray`
  - `Tetrahedron`
  - `Triangle`
- Consolidated `integral(f, ::Geometry, ::GaussKronrod)` code into a single method, eliminating `_integral_gk_*d` and `_gausslegendre` methods.
- Implemented `_zeros` and `_ones` utility functions.
- Implemented `_default_diff_method(geometry)` utility function for specifying default `DifferentiationMethod` per geometry.
- Refactored the unit test system.
  - Standardized `combinations.jl` tests by constructing a `TestableGeometry` package and passing it to a test generation function.
  - Reorganized `@testsnippet`s to exist in same source file as relevant tests.

**Comment by mikeingold** [comment]
@kylebeggs
Cc: @JoshuaLampert 

I think we're in a pretty good position for the upcoming version release. I updated the test suite and left some commented-out provisions for testing the Enzyme backend. Let me know if you have time to re-push `EnzymeExt` or if we should roll it into the next point release.

**Reply by kylebeggs** [reply]
@mikeingold  I could have that done by the end of the week. Feel free to release this now if you wish! I've been traveling this past week so I have not worked on it lately. I should have some time until the end of the year.

**Comment by mikeingold** [comment]
Released!
